zerver/tests/test_signup.py

```diff
@@ -752,3 +753,35 @@ def test_create_realm_with_subdomain(self):
             self.assertEqual(realm.name, realm_name)
             self.assertEqual(realm.subdomain, subdomain)
             self.assertEqual(get_user_profile_by_email(email).realm, realm)
+
+class DeactivateUserTest(ZulipTestCase):
+
+    def test_deactivate_user(self):
+        # type: () -> None
+        email = 'hamlet@zulip.com'
+        self.login(email)
+        user = get_user_profile_by_email('hamlet@zulip.com')
+        self.assertTrue(user.is_active)
+        result = self.client_delete('/json/users/me')
+        self.assert_json_success(result)
+        user = get_user_profile_by_email('hamlet@zulip.com')
+        self.assertFalse(user.is_active)
+        self.login(email, fails=True)
+
+    def test_do_not_deactivate_final_admin(self):
+        # type: () -> None
+        email = 'iago@zulip.com'
+        self.login(email)
+        user = get_user_profile_by_email('iago@zulip.com')
+        self.assertTrue(user.is_active)
+        self.client_delete('/json/users/me')
```

Comment:
for this line, you can use `self.assert_json_error` to confirm that the request will give the expected error.